src/System.Net.Security/src/System/Net/SecureProtocols/SslState.cs

```diff
@@ -596,9 +596,17 @@ internal void ProcessAuthentication(LazyAsyncResult lazyResult)
                         KeyExchangeStrength);
                 }
             }
+            catch (Exception)
+            {
+                // If an exception emerges synchronously, the asynchronous operation was not
+                // initiated, so no operation is in progress.
+                _nestedAuth = 0;
```

Comment:
Might be safer to use InterlockedExchange with the error message from EndAuth. Or is this guaranteed to be atomic?